### PR TITLE
CI: Disable downloading artifacts from upstream jobs

### PR DESCRIPTION
**Pull request by kwryankrattiger**
Setting dependencies to an empty array disables downloading artifacts. Spack CI does not use artifacts between jobs other than those already specifying `needs:artifacts:*` in the parent pipeline.

Ref. https://docs.gitlab.com/ee/ci/yaml/#dependencies

### CODE REVIEW

**Comment by haampie** [comment]
any reason this isn't merged?

**Comment by kwryankrattiger** [comment]
Pending testing on protected pipelines to make sure we will still run signing. Should be done soon.

**Comment by kwryankrattiger** [comment]
Ran this in my experiments and adding the `dependencies: []` does not seem to affect the functionality of pipelines at all

Signing runs: https://gitlab.spack.io/rkratt/pipeline-experiments/-/jobs/9510062
Reindex runs: https://gitlab.spack.io/rkratt/pipeline-experiments/-/jobs/9510063

I think this is good to merge if @haampie or @zackgalbreath want to approve it.

**Comment by kwryankrattiger** [comment]
Actually, forgot about sync jobs for releases/snapshots, tested it here and it is behaving oddly: 
https://gitlab.spack.io/rkratt/pipeline-experiments/-/pipelines/582707

Hold off on clicking merge..

**Comment by kwryankrattiger** [comment]
NVM, turns out I didn't push the tag with the branch, and unknown branch names only run protected publish.

I am going to also add a default of `when: manual` to the base job so that jobs are always present, but marked as manual so the protected publish job does not run erroneously during testing.

Also, now that I tested the copy-job correctly, I found that copy-job has a needs when `artifact_root` is set. This is not compatible with dependencies, and also already blocking the unneeded copying.

Here, there is no mention of that being a problem:
https://docs.gitlab.com/ee/ci/yaml/#dependencies

But if  you read the docs for `needs:artifacts` the explicitly say this is not allowed to mix with `dependencies`: https://docs.gitlab.com/ee/ci/yaml/#needsartifacts

Access was denied because I hadn't setup the protected tags on my project, but the scheduling looks okay which is what really matters for this test.

Here is the CI running with those changes: https://gitlab.spack.io/rkratt/pipeline-experiments/-/pipelines/583132

**Review by zackgalbreath — APPROVED** [review]

**Comment by haampie** [comment]
> The default value for allow_failure is:
>    true for [manual jobs](https://docs.gitlab.com/ee/ci/jobs/job_control.html#create-a-job-that-must-be-run-manually).

This PR seems to cause CI to always pass.

**Comment by kwryankrattiger** [comment]
> This PR seems to cause CI to always pass.

That rule should only apply in the case the branch name doesn't match a pattern name. This should not set the when to manual in all other cases.

**Comment by kwryankrattiger** [comment]
https://gitlab.spack.io/spack/spack/-/pipelines/586182

Despite those not having the manual tag, they have been converted to warnings not failures...that is bad, will revert asap and audit all of the open PRs that benefited from this.

**Pull request merged**
